experimental/corpus/types.go

```diff
@@ -0,0 +1,76 @@
+// Package corpus provides functionality for creating and managing corpora.
+//
+// A corpus is a collection of text documents that are used for training and testing machine learning models.
+// The documents in a corpus are typically sentences or paragraphs of text.
+//
+// The corpus package provides an interface for working with corpora, as well as a set of built-in corpora
+// that can be used for detecting which text will generate false positives in WAF rules.
+//
+// This interface includes methods for getting the URL of the corpus, getting the file from the remote URL,
+// getting an iterator for the corpus, getting the payload given a line from the corpus iterator. Each corpus
+// will have a size, year, source, and language.
+// The iterator interface includes methods for getting the next sentence from the corpus and checking if there
+// is another sentence in the corpus.
+// Each corpus will need its own implementation of the Corpus interface. As this is an experimental package, this
+// interface is subject to change.
+package corpus
+
+// CorpusFile contains the cache directory and file name
+type CorpusFile struct {
```

Comment:
Define a separate interface `CorpusFile`, so implementors of the `CorpusFile` can store other information in the struct. Exporting structs becomes very messy very quickly.